pkg/oauthserver/oauth/handlers/default_auth_handler.go

```diff
@@ -125,10 +124,7 @@ func (authHandler *unionAuthenticationHandler) AuthenticationNeeded(apiClient au
 					// but since we don't accept failedHander here we need to manually alter the audit
 					// event with information about failed authentication
 					ev.ResponseStatus.Message = getAuthMethods(req)
-					ctx = request.WithAuditEvent(ctx, ev)
-					if err := authHandler.requestContextMapper.Update(req, ctx); err != nil {
-						return false, fmt.Errorf("failed to attach audit event to context: %v", err)
-					}
+					req = req.WithContext(request.WithAuditEvent(ctx, ev))
```

Comment:
@liggitt did I get this bit right?

Reply:
I actually don't see why the requestContextMapper.Update call was required before... we're updating a field of a pointer audit event in place. I don't see why the `ev.ResponseStatus.Message = getAuthMethods(req)` call wouldn't be all that is required.

the new request is being propagated anywhere useful, so the `req = req.WithContext(...)` isn't likely to be effective.

@soltysh, where is this data supposed to end up, and how do we tell if it is correct?

Reply:
I think `req = req.WithContext(request.WithAuditEvent(ctx, ev))`

@soltysh, can you describe what we should look for in an audit event to know if this is working correctly with that removed? a failed oauth login attempt should result in an audit event with the auth method included?

Reply:
WithAuditEvent just wraps the context (Golang's linear wrapping chain context design...). So the returned context is a different object than before. You have to attach that to the request again. That's why we had the update.

Reply:
Same is true for the `req = req.WithContext(...)` line now. If you look into WithContext, it does a shallow copy, but does not mutate the old `req`.

Reply:
Or in other words: a context is not a map which is updated in-place. It is a chain of `if key == bla { .... } else ask-the-underlying context` constructs.

Reply:
Right, but we're modifying the existing audit event here, so we shouldn't need to change the context at all

Reply:
Yep, AuditEventFrom is called before. So we change the event in place. Didn't see that from the snippet.